server.js

```diff
@@ -6244,12 +6244,14 @@ cache(function(data, match, sendBadge, request) {
   var url = 'https://hub.docker.com/v2/repositories/' + path + '/stars/count/';
   var badgeData = getBadgeData('docker stars', data);
   request(url, function(err, res, buffer) {
-    if (err != null) {
-      badgeData.text[1] = 'inaccessible';
+    if (checkErrorResponse(badgeData, err, res)) {
       sendBadge(format, badgeData);
       return;
     }
     try {
+      if (isNaN(buffer)) {
```

Comment:
Huh, when is `buffer` NaN?

Let's use [`Number.isNan`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) which was added in ES6 and behaves more expectedly than good old `isNaN`.

Reply:
Unlike the other 3 endpoints/badges, in the valid case, a call like https://hub.docker.com/v2/repositories/library/ubuntu/stars/count/ doesn't return json. It just gives us a single number as the response body. I'm trying to cover the case where we get back a response body that is something else (i.e: the 'unexpected response' case). Usually that would be picked up by `JSON.parse()` throwing an exception but here there is nothing to parse. `Number.isNaN()` isn't the right check because `Number.isNaN("some error message");` returns `false`. Is there a more appropriate check that could be used here?

Reply:
IMO because of the use case it looks like `isNaN()` would be the best option to use here.

Reply:
In the case there’s nothing to parse, do we have an empty buffer? Or undefined?

Reply:
In the 'not found' case, we get back a string error e.g: https://hub.docker.com/v2/repositories/somejunk/not-a-repo/stars/count/

I can't give a concrete example of a situation where we get a 200 OK status code with a body that isn't a number, but I reckon we should attempt to validate the response so that if we did get an unexpected body (be that an error message, an empty body, whatever) with a `200 OK` status, we shouldn't do this:

![docker](https://user-images.githubusercontent.com/6025893/34960269-17286e8c-fa32-11e7-9b31-bcd01a499c52.png)

Reply:
Ah, I gotcha. Hmm. That's an unusual API. Well, we could explicitly validate + convert at the same time, using `parseInt(buffer, 10).